sopel/modules/reddit.py

```diff
@@ -119,14 +120,39 @@ def redditor_info(bot, trigger, match=None):
         client_secret=None,
     )
     match = match or trigger
-    try:  # praw <4.0 style
-        u = r.get_redditor(match.group(2))
-    except AttributeError:  # praw >=4.0 style
+    try:
         u = r.redditor(match.group(2))
-    except Exception:  # TODO: Be specific
+        message = '[REDDITOR] ' + u.name
+        now = dt.datetime.utcnow()
+        cakeday_start = dt.datetime.utcfromtimestamp(u.created_utc)
+        cakeday_start = cakeday_start.replace(year=now.year)
+        day = dt.timedelta(days=1)
+        year_div_by_400 = now.year % 400 == 0
+        year_div_by_100 = now.year % 100 == 0
+        year_div_by_4 = now.year % 4 == 0
+        is_leap = year_div_by_400 or ((not year_div_by_100) and year_div_by_4)
+        if (not is_leap) and ((cakeday_start.month, cakeday_start.day) == (2, 29)):
+            # If cake day is 2/29 and it's not a leap year, cake day is 1/3.
+            # Cake day begins at exact account creation time.
+            is_cakeday = cakeday_start + day <= now <= cakeday_start + (2 * day)
+        else:
+            is_cakeday = cakeday_start <= now <= cakeday_start + day
+
+        if is_cakeday:
+            message = message + ' | ' + bold(color('Cake day', colors.LIGHT_PURPLE))
+        if commanded:
+            message = message + ' | https://reddit.com/u/' + u.name
+        if u.is_gold:
+            message = message + ' | ' + bold(color('Gold', colors.YELLOW))
+        if u.is_mod:
+            message = message + ' | ' + bold(color('Mod', colors.GREEN))
+        message = message + (' | Link: ' + str(u.link_karma) +
+                            ' | Comment: ' + str(u.comment_karma))
+
+        bot.say(message)
+    except prawcore.exceptions.NotFound:
         if commanded:
             bot.say('No such Redditor.')
-            return NOLIMIT
         # Fail silently if it wasn't an explicit command.
         return
```

Comment:
Especially if you're removing the `return NOLIMIT` above, this should not count against rate-limiting.

```suggestion
        return NOLIMIT
```